schemas/tests.yaml

```diff
@@ -0,0 +1,158 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  beakerlib_library:
+    type: array
+    minItems: 1
+
+    items:
+      oneOf:
+        # library(repo/lib)
+        - type: string
+          pattern: "^library([a-zA-Z0-9]+/[a-zA-Z0-9]+)"
+
+        # url: https://github.com/beakerlib/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - url
+
+        # path: /tmp/library/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            path:
+              type: string
+              pattern: "^/.+"
+
+          required:
+            - name
+            - path
+
+        # url: https://github.com/redhat-qe-security/certgen
+        # ref: devel
+        # nick: openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+            nick:
+              type: string
+            ref:
+              type: string
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - nick
+            - ref
+            - url
+
+  string_or_array:
+    oneOf:
+      - type: string
+
+      - type: array
+        minItems: 1
+
+$id: tests.yaml
+
+$schema: https://json-schema.org/draft-07/schema
+
+type: object
+
+additionalProperties: false
+
+properties:
+
+  # TODO: adjust
+
+  component:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  contact:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  description:
+    type: string
+
+  duration:
+    type: string
+    pattern: "^[0-9]+[smhd]$"
+
+  enabled:
+    type: string
+    pattern: "^(true|false)$"
+
+  environment:
+    type: object
+
+    minProperties: 1
+
+    additionalProperties:
+      type: string
+
+  framework:
+    type: string
+    pattern: "^(beakerlib|shell)$"
+
+  # TODO: link
+
+  manual:
+    type: string
+    pattern: "^(true|false)$"
+
+  order:
+    type: integer
+    multipleOf: 1
+
+  path:
+    type: string
+    pattern: "^/.+"
```

Comment:
Maybe something to clarify in the documentation:

> It should be a string containing path from the metadata [tree root](https://tmt.readthedocs.io/en/stable/guide.html#tree) to the desired directory and **should** start with a slash.

Docs say "should" while schema would enforce "must".

Reply:
@psss so?

Reply:
I would say it is a must. The pattern should be `"^/.*"` though as `/` is also a valid path.

Reply:
ack, fixing and I guess I will also udpate the docs

Reply:
updated also docs